src/aquarium-optimized/Context.cpp

```diff
@@ -6,14 +6,52 @@
 
 #include "Context.h"
 
+#include <iostream>
 #include <sstream>
 
+#define GLFW_INCLUDE_NONE
+#include "GLFW/glfw3.h"
+
 #include "imgui.h"
 #include "imgui_impl_glfw.h"
 #include "imgui_internal.h"
 
 #include "Aquarium.h"
 
+int Context::GLFWInitializer::sCounter = 0;
+
+Context::GLFWInitializer::GLFWInitializer() {
+  ++sCounter;
+  glfwInit();  // Initialization failure can be detected with glfwGetError(). On
+               // success, the next glfwInit() invocation will be a no-op.
+}
+
+Context::GLFWInitializer::~GLFWInitializer() {
+  if (--sCounter == 0) {
+    glfwTerminate();  // no effect if GLFW is not initialized
+  }
+}
+
+std::pair<int, int> Context::getMonitorResolution() {
+  GLFWInitializer initializer;
+  if (glfwGetError(nullptr) != GLFW_NO_ERROR) {
+    std::cout << "Failed to initialize GLFW" << std::endl;
+    return {};
+  }
+
+  GLFWmonitor *monitor = glfwGetPrimaryMonitor();
```

Comment:
Can we cache them? (Assume call glfwGetPrimaryMonitor for the assumption that the user have multiple monitors?)

Reply:
This was one of my initial candidate ideas, and I happened to choose the other considering it's only called during initialization. Personally I'd like to avoid introducing too many static variables in the code base, but let's see how others feel about this.